go/multigateway/poolergateway/pooler_gateway.go

```diff
@@ -302,6 +330,43 @@ func (pg *PoolerGateway) Close(ctx context.Context) error {
 // Ensure PoolerGateway implements Gateway
 var _ Gateway = (*PoolerGateway)(nil)
 
+// getPoolerServiceClient returns a MultiPoolerServiceClient for the given database.
+// This can be used for authentication or other admin operations.
+// It finds any available pooler and returns a client connected to it.
+func (pg *PoolerGateway) getPoolerServiceClient(ctx context.Context, database string) (multipoolerpb.MultiPoolerServiceClient, error) {
+	// Find any pooler - for authentication we just need access to pg_authid
+	// which is available from any pooler connected to this database.
+	// Use PRIMARY type as it's always available.
+	target := &query.Target{
+		TableGroup: "default", // TODO: Make configurable or discover from database
+		PoolerType: clustermetadatapb.PoolerType_PRIMARY,
```

Comment:
I'm imagining in the long term that:
- password hashes can come from any primary or replica within a database
- we might get updates through something more like a cache + streaming RPC for invalidation instead of fetching password hashes on demand at connection time

At connection time we don't yet know if the user wants a primary or replica, so assuming primary here is "correct" but until we fix primary routing across zones this could cause issues for people testing in a zone without a primary